src/root.h

```diff
@@ -2,23 +2,22 @@
 #define ROOT_H
 
 #include <memory>
+#include <vector>
 
 #include "child.h"
+#include "clientmanager.h"
+#include "decoration.h"
+#include "hookmanager.h"
+#include "keymanager.h"
+#include "monitormanager.h"
+#include "mousemanager.h"
 #include "object.h"
+#include "rulemanager.h"
+#include "settings.h"
+#include "tagmanager.h"
+#include "tmp.h"
 
-// new object tree root.
-
-class ClientManager;
-class HookManager;
-class KeyManager;
-class MonitorManager;
-class MouseManager;
 class RootCommands;
-class RuleManager;
-class Settings;
-class TagManager;
-class Theme;
-class Tmp;
 
```

Comment:
Here, I disagree. Why do we need to include every manager header in root.h? It looks like IWYU does not understand, that we can define a `Child_<X>` member even if the class `X` is only forward declared.

Reply:
I suppose we can use a pragma to make it understand: https://github.com/include-what-you-use/include-what-you-use/blob/master/docs/IWYUPragmas.md

I haven't looked into it, though.

Reply:
I can't find a pragma that fixes this here. And I'd avoid flooding our code just because IWYU does not get the template right (namely at every occurence of `Child_<X>` where `X` is only forward declared).